Meiliu1313/imp 10 login page

me-liu opened the pull request:
## What's new in this PR

### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- finished fixing styling so there isn't a space + everything lines up
- added colors into styling
- email is removed after input
- string type issues resolved

### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

![image](https://github.com/calblueprint/impact-fund/assets/122945886/122523d3-6c14-4672-b7c3-f9782d0349c7)
![image](https://github.com/calblueprint/impact-fund/assets/122945886/55b3eb07-f75c-4f87-9be3-5971b1d8b1fe)

Old
![image](https://github.com/calblueprint/impact-fund/assets/122945886/5322230b-f864-4c5f-8386-a47fffc13185)
![image](https://github.com/calblueprint/impact-fund/assets/122945886/ce5a4aa0-8bd9-44f1-8640-0ff1c629d0de)
![image](https://github.com/calblueprint/impact-fund/assets/122945886/78065929-7fef-4aae-9d0a-93516cd5b928)

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- there is commented information on the password/index file that is the functionality for the eye icon when it gets updated

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- everything is done except for the eye button in the password + next arrow

## Relevant Links

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

me-liu committed:
screens for email and password login

me-liu committed:
error messages and basic styling

me-liu committed:
error messages and basic styling

me-liu committed:
error messages and basic styling

me-liu committed:
error messages and basic styling

me-liu committed:
error messages and basic styling

me-liu committed:
error messages and basic styling

me-liu committed:
continued styling

me-liu committed:
continued styling

me-liu committed:
continued styling

me-liu committed:
continued styling

me-liu committed:
continued styling

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,61 @@
+  const [displayError, setDisplayError] = useState(false);
+
+  async function emailFind(email: string) {
+    const { data, error } = await supabase
```
A bit of a nit but let's add this supabase query to the case queries file! Let me know if this doesn't makes sense I can explain more in person!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,61 @@
+        </TouchableOpacity>
+      </View>
+
+      <Text style={styles.errorMessage}>
```
Let's add the error message above the next button like the Figma!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,61 @@
+    color: '#920E1A',
+    fontSize: 16,
+    fontWeight: '400',
+    border: 30,
```
`border` isn't a property of the `Text` component so we can remove this!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,61 @@
+    marginBottom: 60,
+    height: 40,
+    margin: 18,
+    borderWidth: 1,
```
Instead of a box, we want a line so we can accomplish this by using `borderBottomWidth` instead of `borderWidth`

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,61 @@
+        onChangeText={setEmail}
+        placeholder="Email address"
+        keyboardType="email-address"
+        autoCapitalize="none"
```
`TextInput` has a `clearButtonMode` field and that can help us accomplish the "x" button!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,60 @@
+import { signInUser } from '../../../../supabase/queries/auth';
+
+export default function LoginScreen() {
+  // const [email, setEmail] = useState('');
```
Remove unused state variable!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,60 @@
+  // const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  const { email } = useLocalSearchParams<{ email: string }>();
```
This is actually a little bit funky and still interprets the email variable as `string | undefined` but Ronnie found a workaround by using `const email = useLocalSearchParams() as unknown as string;`!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,60 @@
+  const [displayError, setDisplayError] = useState(false);
+  const { email } = useLocalSearchParams<{ email: string }>();
+
+  async function signInFunc(email: string, pword: string) {
```
Let's move queries to the auth queries folder!

stephaniewong2 replied:
We can do this by updating the sign in query to return a boolean so that we can use it in this file to determine whether or not to display the error message rather than writing two very similar queries

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,60 @@
+        <Text>Next</Text>
+      </TouchableOpacity>
+
+      <Text style={styles.errorMessage}>
```
Same comment as above but let's move the error message above the next button!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,60 @@
+    marginBottom: 60,
+    height: 40,
+    margin: 18,
+    borderWidth: 1,
```
Same comment as before but let's use `bottomBorderWidth`

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,61 @@
+    const { data, error } = await supabase
+      .from('users')
+      .select()
+      .eq('email', email.toLowerCase());
```
Why do we lower case email?

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,61 @@
+import { Link, router } from 'expo-router';
```
Remove unused `Link` import

stephaniewong2 reviewed (changes requested):
Looks really good!! Functionality is there!! I'll help you with styling and any questions whenever!

me-liu committed:
fixed pr except for eye icon

me-liu committed:
moved queries

me-liu committed:
fixed parenthesis error

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,71 @@
+    setEmail(email);
+  }
+
+  function removeEmail() {
```
Edge case: If a user is typing their email then clears it, the preview text "Email Address" is on both the line and above the text input. We only want this to be on one not both though let's follow up with Josh to see intended functionality!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,75 @@
+    borderBottomWidth: 2,
+    padding: 10,
+    width: 310,
+    borderRadius: 5,
```
Since this is a straight line input we don't need any border radius!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,71 @@
+      </View>
+
+      <View>
+        <TouchableOpacity style={[styles.nextButton]} onPress={() => emailFind}>
```
To call the function we either need to do `onPress={()=>emailFind()}` or `onPress={emailFind}`. Here we can do the latter for cleanliness since it does not take in any parameters

stephaniewong2 replied:
For the "Next" button, let's add styling/logic so that if input is inputted, the button outline/text is black and otherwise it is grey

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,71 @@
+      <View style={styles.errorMessageBox}>
+        <Text style={styles.errorMessage}>
+          {' '}
+          {displayError
```
The text is cut off for me when the error message is displayed! Let's make sure styling accounts for this!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,79 @@
+
+export default function LoginScreen() {
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
```
Same comment as above regarding email but the text is displayed twice when deleting password! Let's fix this after confirming intended functionality with Josh

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,75 @@
+    padding: 10,
+    width: 310,
+    borderRadius: 5,
+    borderColor: '#920E1A',
```
`borderColor` should depend on whether the user is typing or not! It should be red if typing and black otherwise

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,76 @@
+    borderBottomWidth: 2,
+    padding: 10,
+    width: 310,
+    borderRadius: 5,
```
Same comment as above but we don't need a border radius here!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,76 @@
+    padding: 10,
+    width: 310,
+    borderRadius: 5,
+    borderColor: '#920E1A',
```
Same comment as above, but this should depend on whether or not there is text in the input

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,76 @@
+    opacity: 1,
+    fontWeight: '400',
+  },
+  nextButton: {
```
Same comment as above but this should be black if there is input, otherwise grey like you have right now

stephaniewong2 replied:
Resolving because we can ignore for now

stephaniewong2 commented on src/supabase/queries/auth.ts:
```diff
@@ -57,3 +57,27 @@ export const signOutUser = async () => {
+  email: string,
+  password: string,
+): Promise<boolean> => {
+  const { data } = await supabase.auth.signInWithPassword({
```
Here `data` is never null and rather defined like this ```const data: {
    user: User;
    session: Session;
} | {
    user: null;
    session: null;
}``` so let's check for `data.user` instead!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,79 @@
+export default function LoginScreen() {
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  const email = useLocalSearchParams() as unknown as string;
```
This is actually a little funky and assigns `email` to `{"email": "email@email.com"}`. For now, I fixed this by doing `const { email } = useLocalSearchParams<{ email: string }>();
  const userEmail = email as string;` but I will do more research and get back to you if there is a better fix!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,79 @@
+      <View>
+        <TouchableOpacity
+          style={[styles.nextButton]}
+          onPress={() => signInFunc(email, password)}
```
Using what we have above now, let's pass in `userEmail` instead of `email``

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,79 @@
+  const [displayPassword, setDisplayPassword] = useState(false);
+  // const [isPasswordVisible, setPasswordVisible] = useState(false);
+
+  function inputPassword(password: string) {
```
Same little bug/edge case as above but if we type then clear all text, the display is both above and on the line.

stephaniewong2 committed:
[styles] global colors and svgs (#25)

implemented global colors, imported svgs, updated keys so they are preceded with EXPO_PUBLIC_

me-liu committed:
finished everything except svgs and getting button to work depending on if theres input

me-liu committed:
fixed styling bug

me-liu committed:
ok

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+    setIsFocused(false);
+  };
+
+  function inputEmail(email: string) {
```
Let's remove this!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+      setDisplayError(true);
+    } else {
+      setDisplayError(false);
+      if (typeof email === 'string') {
```
I don't think we need this check, we can just navigate to the next screen because we only accept text/string input

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+      <TextInput
+        style={[styles.input, isFocused && styles.inputFocused]}
+        value={email}
+        onChangeText={inputEmail}
```
Let's call `setEmail` here instead

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+        <TouchableOpacity
+          style={[styles.nextButton]}
+          onPress={emailFind}
+          // disabled={!isEmail}
```
Remove if unused

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
Overall styling comments: let's make sure everything is aligned -- "Please enter...", "Email address", text input, error message, next button. Josh also wants the input line before inputting/empty input (in black) to be thinner but when text is inputted leave as is. Besides that, styling is good from my end. Josh will also look soon!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+      setDisplayError(false);
+      if (typeof email === 'string') {
+        router.push({ pathname: 'Login/Password', params: { email } });
+      }
```
Let's set the `email` state variable to an empty string again after this!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+export default function LoginScreen() {
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  // const { email } = useLocalSearchParams<{ email: string }>();
```
Let's do `const { email } = useLocalSearchParams() as unknown as { email: string };`

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+  const [password, setPassword] = useState('');
+  const [displayError, setDisplayError] = useState(false);
+  // const { email } = useLocalSearchParams<{ email: string }>();
+  // const userEmail = email as string;
```
Remove commented out code!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+      setDisplayPassword(false);
+    }
+    setPassword(password);
+    /* if (password.trim() !== '') {
```
Remove if not used

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+    }
+  }
+  /*
+  const togglePasswordVisibility = () => {
```
Remove if unused -- we can bring this back when we handle the eye icon

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  async function signInFunc() {
+    /*
+    if (typeof email !== 'string') {
```
remove

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+  };
+  */
+
+  async function signInFunc() {
```
Nit, but let's call this `signIn` instead of `signInFunc`

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
Same styling comments as login screen

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+  // const userEmail = email as string;
+  const email = useLocalSearchParams<{ email: string }>();
+  const [displayPassword, setDisplayPassword] = useState(false);
+  // const [isPassword, setIsPassword] = useState(false);
```
Remove if unused

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,99 @@
+        <Text style={styles.errorMessage}>
+          {' '}
+          {displayError
+            ? 'The email you entered is either incorrect or not registered with Impact Fund.'
```
There is a space before the first word in the error message -- " The email..." instead of "The email"

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,108 @@
+        <Text style={styles.errorMessage}>
+          {' '}
+          {displayError
+            ? 'Oh no! The password you entered is incorrect, please try again.'
```
There is a space before the first word in the error message -- " Oh no..." instead of "Oh no"

me-liu committed:
fixed comments and styled so that everything was aligned

me-liu committed:
finished changes from pr

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,90 @@
+
+      <View style={styles.errorMessageBox}>
+        <Text style={styles.errorMessage}>
+          {' '}
```
I think this is causing the weird indent so we can remove this!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,90 @@
+        <Text style={styles.errorMessage}>
+          {' '}
+          {displayError
+            ? 'The email you entered is either incorrect or not \n registered with Impact Fund.'
```
Let's not add a new line in between because it is dependent on the screen itself

stephaniewong2 commented on src/app/(Authentication)/Login/Email/index.tsx:
```diff
@@ -0,0 +1,90 @@
+        clearButtonMode="while-editing"
+      />
+
+      <View style={styles.errorMessageBox}>
```
I believe if you get rid of this View and update styling a bit to account for the button, the text will be properly aligned.

stephaniewong2 commented on src/app/(Authentication)/Login/Password/index.tsx:
```diff
@@ -0,0 +1,89 @@
+
+      <View style={styles.errorMessageBox}>
+        <Text style={styles.errorMessage}>
+          {' '}
```
Same comments as Email!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,92 @@
+    marginTop: 5,
+    height: 40,
+    borderBottomWidth: 1.5,
+    padding: 10,
```
Remove padding!

stephaniewong2 replied:
Remove padding!

stephaniewong2 commented on src/app/(Authentication)/Login/Email/styles.ts:
```diff
@@ -0,0 +1,92 @@
+    marginTop: 5,
+    height: 40,
+    borderBottomWidth: 2,
+    padding: 10,
```
Remove padding!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,94 @@
+    marginTop: 5,
+    height: 40,
+    borderBottomWidth: 1.5,
+    padding: 10,
```
Remove padding!

stephaniewong2 commented on src/app/(Authentication)/Login/Password/styles.ts:
```diff
@@ -0,0 +1,94 @@
+    marginTop: 5,
+    height: 40,
+    borderBottomWidth: 2,
+    padding: 10,
```
Remove padding!

philipye314 committed:
[Cases] Eligibility Form and Status

Did Eligibility Screen, EligibilityForm. Updates specific user/case on button press.

stephaniewong2 committed:
[fix] use supabase storage for images 

* images from supabase storage

* remove image field from case type -- using case id instead

* delete comment

* move to queries folder

* cleanup

* type state variable, fixed warning

* [Cases] Eligibility Form and Status

Did Eligibility Screen, EligibilityForm. Updates specific user/case on button press.

* images from supabase storage

* remove image field from case type -- using case id instead

* delete comment

* move to queries folder

* cleanup

* type state variable, fixed warning

* fix

---------

Co-authored-by: Philip Ye <97428041+philipye314@users.noreply.github.com>

me-liu committed:
basically done

ronniebeggs committed:
[cases] case summary component (#22)

* [feat] build base layout.

* [feat] style component with hardcoded data.

* [feat] conditional render case data; fix styling.

* [feat] add navigation to CaseSummaryScreen page.

* [feat] pass case data into case summary screen.

* [style] section off information by type.

* [style] continue styling to match figma.

* [wip][feat] implement base external link component.

* [feat] add routing to link within external browser.

* [feat] add header component with impact fund logo to all Cases related screens.

* [cleanup] final cleanup before PR.

* [fix] fix rendering issue on the bottom of CaseSummaryScreen.

* [feat] implement date format util.

* [style] restyle sub text to better reflect figma designs.

* [feat] build base layout.

* [feat] style component with hardcoded data.

* [feat] conditional render case data; fix styling.

* [feat] add navigation to CaseSummaryScreen page.

* [feat] pass case data into case summary screen.

* [style] section off information by type.

* [style] continue styling to match figma.

* [wip][feat] implement base external link component.

* [feat] add routing to link within external browser.

* [feat] add header component with impact fund logo to all Cases related screens.

* [cleanup] final cleanup before PR.

* [fix] fix rendering issue on the bottom of CaseSummaryScreen.

* [feat] implement date format util.

* [style] restyle sub text to better reflect figma designs.

* [style] change div dimensions to percentages/aspect ratios.

* [style] implement global colors.

* [style] final styling changes + shadows.

* [fix] prevent eslint error.

* [style] replace react icons with svgs.

* [refactor] restructure image fetch from supabase storage; redo typing case-related data.

* [wip]

* [cleanup] minor query refactoring and style fixes.

* [style] setup scroll view container for CaseScreen.

* [cleanup] fix styling.

me-liu committed:
changes after rebase

me-liu committed:
more rebasing

me-liu committed:
commiting before rebase
